### PR TITLE
chore: prep for move to procore-oss github organization

### PR DESCRIPTION
**Pull request by jmeridth**
- [x] Set default/lowest version to node 18 [LTS ends in a year](https://endoflife.date/nodejs)
- [x] setup .github folder (actions, dependabot.yaml, ISSUE_TEMPLATES, PULL_REQUEST_TEMPLATE.md, CODEOWNERS)
- [x] remove circleci, replace with github actions
- [x] setup required files (README.md, SECURITY.md, CODE_OF_CONDUCT.md, LICENSE.md, CONTRIBUTING.md)
- [x] update npm information
- [x] verify no procore specific stuff present
- [x] README markdown linting fixes

### CODE REVIEW

File: .github/CODEOWNERS
Comment (jmeridth) [inline .github/CODEOWNERS:3]: This will be correct once we move the repo over to github.com/procore-oss GitHub organization.

**Comment by jmeridth** [comment]
Plan:

- merge this PR once approved
- I put in CSEIN ticket to move repo to github.com/procore-oss organization
- I setup new NPM_TOKEN and prep release
- increase version and do release to `procore/js-sdk npm package`

**Review by njbbaer — APPROVED** [review]

**Pull request merged**
